File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        longest = 0
        for i in s:
            if i-1 in s:
                continue
            length = 1
            n = i
            while n+1 in s:
                length += 1
                n += 1

            longest = max(longest, length)
        return longest

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size();
        if (n == 0) return 0;
        int back = 0;
        int max_length = 1;
        unordered_set<char> se;
        se.insert(s[0]);
        for (int front = 1; front < n; front++){
            if (se.contains(s[front])){
                max_length = max(max_length, front-back);
                while (se.contains(s[front])){
                    se.erase(s[back]);
                    back++;
                }

            }
            se.insert(s[front]);
            
            
        }

        max_length = max(max_length, n-back);

        return max_length;
    }
};
